vendor/src/github.com/docker/libnetwork/sandbox_store.go

```diff
@@ -132,7 +134,16 @@ func (sb *sandbox) storeUpdate() error {
 		sbs.Eps = append(sbs.Eps, eps)
 	}
 
-	return sb.controller.updateToStore(sbs)
+	err := sb.controller.updateToStore(sbs)
+	if err == datastore.ErrKeyModified {
+		// When we get ErrKeyModified it is sufficient to just
+		// go back and retry.  No need to get the object from
+		// the store because we always regenerate the store
+		// state from in memory sandbox state
+		goto retry
```

Comment:
It depends on how many go routines are racing at any point in time. If there are 10 racing go routines then a particular unfortunate go routine may retry a max of 10 times. The worst case number of retries by any one go routine increases linearly with the number of racing go routines. But still finite.